pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java

```diff
@@ -153,6 +163,10 @@ public SchemaInfo getSchemaInfo(byte[] schemaVersion) {
         return null;
     }
 
+    public SchemaInfo getAutoConsumeSchemaInfo() {
```

Comment:
You can of course create a SchemaInfo here.
https://github.com/apache/pulsar/pull/17449/files#r1061103612
But I feel like giving `AutoConsumeSchema` a "default" schemaInfo. Using it in `ConsumerImpl` will make the code a little cleaner. What do you think of my thoughts here.

Reply:
It's not good to use a non-static method to return a static variable, like the following code:

```java
class Foo {
    private static String name = "Foo";
    public String getName() { return name; }
}
// Users have to call `foo.getName()` for a Foo instance `foo`, but
// it's better to add a static method like `Foo.getName()`.
```

Back to the code here, it's redundant to add a method to get the schema info. Why not just make `SCHEMA_INFO` public and reference `AutoConsumerSchema.SCHEMA_INFO`?

```java
    public static final SchemaInfo SCHEMA_INFO = SchemaInfoImpl.builder()
                .name("AutoConsume")
                .type(SchemaType.AUTO_CONSUME)
                .schema(new byte[0])
                .build();
```

```java
            if (schema instanceof AutoConsumeSchema
                    && Commands.peerSupportsCarryAutoConsumeSchemaToBroker(cnx.getRemoteEndpointProtocolVersion())) {
                si = AutoConsumeSchema.SCHEMA_INFO;
            }
```

I don't think it would be less clean than the code in your PR.

Reply:
That is really a very good suggestion. I made a mistake here and I will change it to your design.